### PR TITLE
adds correlation id to log context

### PR DESCRIPTION
**Pull request by engenb**
Just wanted to propose this as a POC before taking it further...

What you're doing here is very similar to something I'm doing as well, but yours is better.  If you can't beat 'em, join em!

The only thing missing from what I'd like is for the correlation id to be added to the log context to easily group log entries in structured log aggregators.

Requires an additional dependency on Microsoft.Extensions.Logging.Abstractions but it's a pretty safe bet that's not an issue since this is intended for use with AspNetCore.

Thoughts?

### CODE REVIEW

File: src/CorrelationId/CorrelationIdMiddleware.cs [hunk 1]
```diff
@@ -55,7 +59,13 @@ public async Task Invoke(HttpContext context, ICorrelationContextFactory correla
                 });
             }
 
-            await _next(context);
+            using (_logger.BeginScope(new Dictionary<string, object>
+            {
+                {nameof(CorrelationContext.CorrelationId), correlationId}
```
Comment (SmilieLyon) [inline src/CorrelationId/CorrelationIdMiddleware.cs]: This is incorrect, you should use  _options.Header instead

Comment (SmilieLyon) [reply src/CorrelationId/CorrelationIdMiddleware.cs]: Unless you want to add in another variable variable and track it through? 
Having the flexibility of being able to change the logging name would be advantageous.

**Comment by SmilieLyon** [comment]
I love the idea and was planning on making a similar PR.
My company however wants to have a different logging property name than "CorrelationId"

File: src/CorrelationId/CorrelationIdMiddleware.cs (hunk 1 above)
Comment (stevejgordon) [reply src/CorrelationId/CorrelationIdMiddleware.cs]: Agree, being able to configure this makes sense.

**Pull request merged**